packages/flutter/lib/src/widgets/editable_text.dart

```diff
@@ -1792,6 +1793,15 @@ class EditableText extends StatefulWidget {
   /// {@endtemplate}
   bool get selectionEnabled => enableInteractiveSelection;
 
+  /// {@template flutter.widgets.editableText.highlightAllOnFocus}
+  /// Whether or not this field should highlight all text when gaining focus on
+  /// web or desktop
```

Comment:
The special case for web and desktop makes me hesitate here. I know that's now it works now, though. What happens in a native iOS or Android app if you connect a hardware keyboard and use the tab key to move between fields?